### DISCUSSION TITLE
Thumb cluster screw hole generated outside when choosing inside screw holes

### DISCUSSION THREAD

**Discussion by fengzixiaopi** [joshreve/dactyl-keyboard#26] [General]
Hi,
I tried to generate a 4x6 one with inside screw holes but the hole blow thumb cluster was generated outside instead. Is this expected behavior to avoid blocking some thumb keys or a bug? Thanks!

![Capture](https://user-images.githubusercontent.com/68142426/128914645-04706217-5dce-4eeb-af0b-68c0426e2b07.PNG)

**Comment by joshreve** [comment]
Nice find.  I will fix it when I merge the trackball branch, likely this weekend, but here is the quick fix until then:

```
def screw_insert_thumb(bottom_radius, top_radius, height):
    if thumb_style == 'MINI':
        position = thumborigin()
        position = list(np.array(position) + np.array([-29, -51, -16]))
        position[2] = 0

```
LINE BELOW WAS ORIGINALLY:
```
if thumb_style == 'MINIDOX':
        position = thumborigin()
        position = list(np.array(position) + np.array([-37, -32, -16]))
```
LINE BELOW SHOULD BE:
```
    elif thumb_style == 'MINIDOX':
        position = thumborigin()
        position = list(np.array(position) + np.array([-37, -32, -16]))

```
It should be a chain of if, elif, elif, else.  I missed the second elif triggering the else at the end.  I think I reordered it at one point and messed it up.

**Reply by fengzixiaopi** [reply]
Thanks!
